### PR TITLE
CLN: Remove getitem special error behaviour

### PR DESCRIPTION
**Pull request by m-richards**
Addresses https://github.com/geopandas/geopandas/issues/2517

Based on discussion in #2857, remove the special None handling of geometry column name as the associated error message is less helpful than the equivalent for loc/iloc.

This includes #2857, so will have a much smaller diff after that is merged.

The end goal is to simplify the logic in #2577 by removing `_geometry_column=None` as a possible case.

### CODE REVIEW

**Review by martinfleis — APPROVED** [review]
Looks good to me. Thanks!

File: geopandas/tests/test_op_output_types.py [hunk 1]
```diff
@@ -238,12 +226,13 @@ def identity(x):
...
     assert_object(df[[geo_name, "geometry2"]].apply(identity), GeoDataFrame, geo_name)
     assert_object(df[[geo_name]].apply(identity), GeoDataFrame, geo_name)
-    expected_geo_col_name = None if compat.PANDAS_GE_14 else "geometry"
-    assert_obj_no_active_geo_col(
-        df[["geometry2", "value1"]].apply(identity), GeoDataFrame, expected_geo_col_name
-    )
+    geo_name_none = geo_name if compat.PANDAS_GE_14 else "geometry"
+
+    res = df[["geometry2", "value1"]].apply(identity)
+    assert res._geometry_column_name == geo_name_none
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_op_output_types.py:232]: This assert is also done in the `assert_obj_no_active_geo_col` on the line below?

File: geopandas/tests/test_op_output_types.py [hunk 2]
```diff
@@ -238,12 +226,13 @@ def identity(x):
     assert_object(df[["value1", "value2"]].apply(identity), pd.DataFrame)
     assert_object(df[[geo_name, "geometry2"]].apply(identity), GeoDataFrame, geo_name)
     assert_object(df[[geo_name]].apply(identity), GeoDataFrame, geo_name)
-    expected_geo_col_name = None if compat.PANDAS_GE_14 else "geometry"
-    assert_obj_no_active_geo_col(
-        df[["geometry2", "value1"]].apply(identity), GeoDataFrame, expected_geo_col_name
-    )
+    geo_name_none = geo_name if compat.PANDAS_GE_14 else "geometry"
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_op_output_types.py:229]:
```suggestion
    expected_geo_col_name = geo_name if compat.PANDAS_GE_14 else "geometry"
```

the "none" is a bit confusing?

**Review by jorisvandenbossche — APPROVED** [review]
Just some minor comment on the tests, but go ahead and merge when ready!

**Pull request merged**